clients/drcachesim/tools/invariant_checker.cpp

```diff
@@ -696,7 +710,7 @@ invariant_checker_t::parallel_shard_memref(void *shard_data, const memref_t &mem
                 // signal should be same as the pre-signal instr for the first one.
                 // Here we restore last_instr_in_cur_context_ to the last instr we
                 // saw *in the same nesting depth* before the first signal.
-                shard->last_instr_in_cur_context_ =
+                shard->last_instr_in_cur_context_.instr =
```

Comment:
But this needs to set the other fields, or invalidate them.  pre_signal_instr should probably be an instr info type.